File: app/src/main/java/com/example/animabuilder/CharacterCreation/Attributes/RaceName.java
package com.example.animabuilder.CharacterCreation.Attributes;

import java.io.Serializable;

public enum RaceName implements Serializable {
    human,
    sylvain,
    jayan,
    danjayni,
    ebudan,
    daimah,
    dukzarist;

    public static RaceName fromString(String input){
        switch(input){
            case "sylvain":
            case "Sylvain": return sylvain;

            case "jayan":
            case "Jayan": return jayan;

            case "danjayni":
            case "D\'Anjayni": return danjayni;

            case "ebudan":
            case "Ebudan": return ebudan;

            case "daimah":
            case "Daimah": return daimah;

            case "dukzarist":
            case "Duk\'zarist": return dukzarist;

            case "human":
            case "Human":
            default: return human;
        }
    }
}

File: app/src/main/java/com/example/animabuilder/CharacterCreation/Attributes/RaceName.kt
package com.example.animabuilder.CharacterCreation.Attributes

import com.example.animabuilder.CharacterCreation.Attributes.RaceName
import java.io.Serializable

enum class RaceName : Serializable {
    human, sylvain, jayan, danjayni, ebudan, daimah, dukzarist;

    companion object {
        @JvmStatic
        fun fromString(input: String?): RaceName {
            return when (input) {
                "sylvain", "Sylvain" -> sylvain
                "jayan", "Jayan" -> jayan
                "danjayni", "D\'Anjayni" -> danjayni
                "ebudan", "Ebudan" -> ebudan
                "daimah", "Daimah" -> daimah
                "dukzarist", "Duk\'zarist" -> dukzarist
                "human", "Human" -> human
                else -> human
            }
        }
    }
}